pallets/token-manager/src/test_deferred_lower.rs

```diff
@@ -0,0 +1,287 @@
+
+        ext.execute_with(|| {
+            let (_, from, _, t1_recipient) = MockData::setup_lower_request_data();
+            let pre_lower_balance = <TokenManager as Store>::Balances::get((NON_AVT_TOKEN_ID, from));
+            let amount = pre_lower_balance;
+
+            let expected_lower_id = 0;
+            let expected_schedule_name = ("Lower", &expected_lower_id).using_encoded(sp_io::hashing::blake2_256);
+            let expected_execution_block = get_expected_execution_block();
+
+            schedule_lower(from, amount, t1_recipient);
+
+            // The lower has been successfully scheduled in the scheduler pallet
+            assert_eq!(1, Agenda::<TestRuntime>::get(expected_execution_block).len());
+
+            // Event emitted
+            assert!(System::events().iter().any(|a| a.event ==
+                RuntimeEvent::TokenManager(crate::Event::<TestRuntime>::LowerRequested {
+                    token_id: NON_AVT_TOKEN_ID,
+                    from,
+                    amount,
+                    t1_recipient,
+                    sender_nonce: None,
+                    lower_id: expected_lower_id,
+                    schedule_name: expected_schedule_name,
+                }))
+            );
+
+            // Cancel the lower
+            assert_ok!(Scheduler::cancel_named(RuntimeOrigin::root(), expected_schedule_name));
+
+            // move to the expected execution block
+            fast_forward_to_block(expected_execution_block);
+
+            // No tokens have been burned even when the expected execution block is reached
+            assert_eq!(
+                <TokenManager as Store>::Balances::get((NON_AVT_TOKEN_ID, from)),
+                pre_lower_balance
+            );
+
+            // The lower has been removed from the scheduler pallet
+            assert_eq!(0, Agenda::<TestRuntime>::get(expected_execution_block).len());
+        });
+    }
+
+    #[test]
+    fn non_root_cancellation_fails() {
+        let mut ext = ExtBuilder::build_default().with_genesis_config().as_externality();
+
+        ext.execute_with(|| {
+            let (_, from, _, t1_recipient) = MockData::setup_lower_request_data();
+            let pre_lower_balance = <TokenManager as Store>::Balances::get((NON_AVT_TOKEN_ID, from));
+            let amount = pre_lower_balance;
+
+            let expected_lower_id = 0;
+            let expected_schedule_name = ("Lower", &expected_lower_id).using_encoded(sp_io::hashing::blake2_256);
+            let expected_execution_block = get_expected_execution_block();
+
+            schedule_lower(from, amount, t1_recipient);
+
+            // The lower has been successfully scheduled in the scheduler pallet
+            assert_eq!(1, Agenda::<TestRuntime>::get(expected_execution_block).len());
+
+            // Only root can cancel the lower
+            assert_noop!(Scheduler::cancel_named(RuntimeOrigin::signed(from), expected_schedule_name), DispatchError::BadOrigin);
```

Comment:
do we need this test since the `Scheduler` pallet works with sudo extrinsics only?

Reply:
The schedule can be configured to accept non sudo transactions using its config traits.